app/controllers/users.py

```diff
@@ -893,3 +895,67 @@ def get(self):
             ), 200
         except Exception as err:
             return dict(status='fail', message=str(err)), 400
+
+class InActiveUsersView(Resource):
+    computed_results = {}  # Dictionary to cache computed results
+    
+    @admin_required
+    def get(self):
+        user_schema = UserSchema(many=True)
+        page = request.args.get('page', 1,type=int)
+        per_page = request.args.get('per_page', 10, type=int)
+        entered_date = request.args.get("date")
+
+        today = datetime.now().date()
+
+        if (entered_date is not None):
+            try:
+                entered_date = datetime.strptime(entered_date, "%Y-%m-%d").date()  # Standardize the date format
+            except ValueError:
+                return dict(status='fail', message="Invalid date format"), 400
+        else:
+            return dict(status='fail', message="Missing required parameters"), 400
+        
+        if entered_date > today:
+                return dict(status='fail', message="Entered date cannot be in the future"), 400
+
+        time_difference = relativedelta(today, entered_date)
+        days_difference = time_difference.days
+
+        if days_difference in self.computed_results:
+            users_data = self.computed_results[days_difference]
+            total_items = len(json.loads(users_data))
+            total_pages = ceil(total_items / per_page)
+
+            pagination = {
+                'total': total_items,
+                'pages': total_pages,
+                'page': page,
+                'per_page': per_page,
+                'next': page + 1 if page < total_pages else None,
+                'prev': page - 1 if page > 1 else None
+            }
```

Comment:
```suggestion
            pagination , users_data = paginate(self.computed_results,per_page,page)
```
We have a helper function for pagination under the helper functions that we can reuse in such cases of manual pagination . It helps keep the code more precise, easy to read and maintainable.
Import it from `from app.helpers.pagination import paginate`